src/in_progress.rs

```diff
@@ -1110,3 +1110,138 @@ pub fn is_hardware_waiting_repair(
     let stmt_query = conn.query(&stmt, &params)?;
     Ok(!stmt_query.is_empty())
 }
+
+/// Get region id based on the region name.
+pub fn get_region_id(pool: &Pool<ConnectionManager>, region_name: &str) -> BynarResult<u32> {
+    let conn = get_connection_from_pool(pool)?;
+
+    // Get region Id from region name
+    let stmt = format!(
+        "SELECT region_id FROM regions WHERE region_name = '{}'",
+        region_name
+    );
+    let stmt_query = conn.query(&stmt, &[])?;
+
+    let mut region_id: u32 = 0;
+
+    if let Some(res) = stmt_query.into_iter().next() {
+        // Exists, return region_id
+        let id: i32 = res.get(0);
+        region_id = id as u32;
+        debug!("Region id {} for the region {}", region_id, region_name);
+    } else {
+        // does not exist
+        debug!("No region with name {} in database", region_name);
```

Comment:
The logic here picks up on the possibility that region_name might not exist in the database.  We can improve this function by changing the type signature of the function to indicate that this is possible as well:
`pub fn get_region_id(pool: &Pool<ConnectionManager>, region_name: &str) -> BynarResult<Option<u32>>`
This says it's possible that this function won't find a region_id
Now in your if/else block here you could return `Ok(Some(region_id))` or if it can't be found you return `Ok(None)`

Reply:
Fixed, tested and committed